include/deal.II/fe/fe_poly.h

```diff
@@ -478,10 +482,9 @@ class FE_Poly : public FiniteElement<dim, spacedim>
     const unsigned int dof) const;
 
   /**
-   * The polynomial space. Its type is given by the template parameter
-   * PolynomialType.
+   * The polynomial space.
    */
-  PolynomialType poly_space;
+  std::unique_ptr<ScalarPolynomialsBase<dim>> poly_space;
```

Comment:
Might also want to do here as I do in #8716.

Reply:
Got it on my end!